### PR TITLE
Fix error in parallelization scheme

### PR DESCRIPTION
**Pull request by marinegor**
Changes made in this Pull Request:
 - fix figure as mentioned in #4729 discussion
 - updates docs accordingly
 - additionally mentions progress about parallelization implementation in docs

PR Checklist
------------
 - [ ] Tests?
 - [x] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

## Developers certificate of origin
- [ ] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4760.org.readthedocs.build/en/4760/

### CODE REVIEW

File: package/doc/sphinx/source/documentation_pages/analysis/parallelization.rst [hunk 1]
```diff
@@ -41,7 +41,8 @@ impossible with any but the ``serial`` backend.
 
    Parallelization is getting added to existing analysis classes.  Initially,
    only :class:`MDAnalysis.analysis.rms.RMSD` supports parallel analysis, but
-   we aim to increase support in future releases.
+   we aim to increase support in future releases. Please check issues labeled
+   `parallelization` on the MDAnalysis issues [page](https://github.com/MDAnalysis/mdanalysis/issues).
```
Comment (orbeckst) [inline package/doc/sphinx/source/documentation_pages/analysis/parallelization.rst]: You can make the link https://github.com/MDAnalysis/mdanalysis/issues?q=is%3Aopen+is%3Aissue+label%3Aparallelization

**Review by orbeckst — APPROVED** [review]
Good fixes to the docs. Important for anyone wanting to work with parallelization.

@IAlibay this is not a new feature but a docs bug fix. Can this go in despite feature freeze?

**Comment by orbeckst** [comment]
@IAlibay I assigned the PR to you so you can make the decision about merging despite 2.8.0 feature freeze.

**Review by yuxuanzhuang — APPROVED** [review]
Thanks! lgtm!

File: package/doc/sphinx/source/documentation_pages/analysis/parallelization.rst (hunk 1 above)
Comment (IAlibay) [inline package/doc/sphinx/source/documentation_pages/analysis/parallelization.rst]:
```suggestion
   we aim to increase support in future releases. Please check issues labeled
   `parallelization` on the [MDAnalysis issues tracker](https://github.com/MDAnalysis/mdanalysis/issues?q=is%3Aopen+is%3Aissue+label%3Aparallelization).
```

File: package/doc/sphinx/source/documentation_pages/analysis/parallelization.rst [hunk 2]
```diff
@@ -41,7 +41,8 @@ impossible with any but the ``serial`` backend.
 
    Parallelization is getting added to existing analysis classes.  Initially,
    only :class:`MDAnalysis.analysis.rms.RMSD` supports parallel analysis, but
-   we aim to increase support in future releases.
+   we aim to increase support in future releases. Please check issues labeled
+   `parallelization` on the [MDAnalysis issues tracker](https://github.com/MDAnalysis/mdanalysis/issues?q=is%3Aopen+is%3Aissue+label%3Aparallelization).
```
Comment (IAlibay) [inline package/doc/sphinx/source/documentation_pages/analysis/parallelization.rst]:
```suggestion
   `parallelization` on the `MDAnalysis issues tracker <https://github.com/MDAnalysis/mdanalysis/issues?q=is%3Aopen+is%3Aissue+label%3Aparallelization>`_.
```
sphinx not markdown

**Review by IAlibay — APPROVED** [review]
I went ahead and fixed the link so we can merge more rapidly.

**Pull request merged**